### PR TITLE
ci: add spellcheck for messages module and checklist for messages changes

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Why is this needed?

This adds CI with two things:
- spell check
- a new comment to prompt for explanation when translatable messages are being changed. This will continuously remind us of our change's impact on translations, but hopefully won't actually introduce too much friction into our development processes. This PR modified messages, so this PR is a good example of what extra process is required from this new comment.

## Test Steps

No functional changes to pro-client.

## Checklist

 - [n/a] I have updated or added any unit tests accordingly
 - [n/a] I have updated or added any integration tests accordingly
 - [x] Changes here need to be documented, and this was done in: #2777

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

**Comment by orndorffgrant** [comment]
> ⚠️ Please add a comment with justification of why messages are being altered.

The changes are only typo fixes and existing translations are preserved.

File: .github/actions/checklists/action.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: 'Custom Checklists'
+description: 'Add context specific checklists to the PR'
```
Comment (s-makin) [inline .github/actions/checklists/action.yml:2]:
```suggestion
description: 'Add context-specific checklists to the PR'
```

File: .github/actions/checklists/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    const filenames = files.map(f => f.filename);
+
+    const modifiesMessages = filenames.includes("uaclient/messages/__init__.py")
+    const theComment = comments.data.find(c => c.body.includes(messageChangesCommentHeader));
+
+    console.log({modifiesMessages, commentExists: !!theComment});
+
+    if (theComment && !modifiesMessages) {
+        console.log("comment already exists, but PR no longer modifies messages, delete it")
```
Comment (s-makin) [inline .github/actions/checklists/index.js:55]: On this line and 62, can we change these to "delete comment" and "create comment" respectively? (the "it" on both lines is ambiguous)

File: .github/actions/checklists/package.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "checklists",
+  "version": "1.0.0",
+  "description": "Add context specific checklists to the PR",
```
Comment (s-makin) [inline .github/actions/checklists/package.json:4]:
```suggestion
  "description": "Add context-specific checklists to the PR",
```

**Review by s-makin — APPROVED** [review]
Looks good to me, just a couple of tiny nits. One question - is the spellcheck able to be run locally? (if so, do we have instructions on that somewhere, or is it automatic?)

**Comment by orndorffgrant** [comment]
Thanks @s-makin! I've addressed your inline comments here. And I'll add a short how-to to #2777 on how to run spellcheck locally

**Review by s-makin — APPROVED** [review]
Thanks for the changes :) lgtm

**Review by renanrodrigo — APPROVED** [review]

**Pull request merged**